File: llama3/sistemaDeTipos/tipos.py
def promedio_sobre_umbral(precios, umbral):
    # Comenzamos con un valor inicial de 0
    total = 0
    count = 0
    
    # Iteramos sobre la lista original sin modificarla
    for precio in precios[:]:
        if precio > umbral:
            # Aumentamos el total y el conteo si el precio es mayor al umbral
            total += precio
            count += 1
    
    # Si no hay elementos que superen el umbral, devolvemos 0
    if count == 0:
        return 0
    
    # Calculamos el promedio y lo devolvemos
    return total / count

# Ejemplo de uso
precios = [10, 20, 30, 40, 50]
promedio = promedio_sobre_umbral(precios, 25)

print(promedio)  # 40.0

File: llama3/sistemaDeTipos/tipos.c
#include <stdio.h>
#include <stdlib.h>

double calcularPrecioPromedioMayorUMbral(double* precios, int tamano, double umbral) {
    if (tamano == 0) return 0; // lista vacía

    double suma = 0;
    int cuenta = 0;

    for (int i = 0; i < tamano; i++) {
        if (precios[i] > umbral) {
            suma += precios[i];
            cuenta++;
        }
    }

    return cuenta == 0 ? 0 : suma / cuenta;
}

int main() {
    double precios[] = {10.0, 20.0, 30.0, 40.0, 50.0};
    int tamano = sizeof(precios) / sizeof(precios[0]);
    double umbral = 20;

    printf("Precio promedio mayor al umbral: %.2f\n", calcularPrecioPromedioMayorUMbral(precios, tamano, umbral));

    return 0;
}